std-bits/base/src/main/java/org/enso/base/encoding/ReportingStreamDecoder.java

```diff
@@ -80,14 +98,28 @@ public ReportingStreamDecoder(InputStream stream, CharsetDecoder decoder) {
   private boolean hadEofDecodeCall = false;
 
   /**
-   * A list of positions containing encoding issues like malformed characters.
+   * Our read method tries to read as many characters as requested by the user.
    *
-   * <p>Used for reporting warnings.
+   * <p>It will never return 0 - it will either return a positive number of characters read, or -1
```

Comment:
This comment seems to contradict line 120

Reply:
Ooops! I forgot about this edge case, I will update the doc 'will never return 0 unless user requested 0 bytes'. As the user setting `len == 0` should be the _only_ case where `0` would be returned - as guaranteed by asserts below.